extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts

```diff
@@ -0,0 +1,174 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { ProxyIdentifier, createProxyIdentifier } from '@theia/plugin-ext/lib/api/rpc-protocol';
+
+export interface CheApiPlugin {
+
+}
+
+export interface CheApiMain {
+    $currentWorkspace(): Promise<Workspace>;
+
+    $getFactoryById(id: string): Promise<Factory>;
+}
+
+export interface Factory {
```

Comment:
why Factory, FactoryAction, etc interfaces are defined there ?
it should be just imports from d.ts file no ?

Reply:
They not the same

Reply:
if we can try to have prefix name to differentiate them
I think it would help for code readability.